### PR TITLE
feat(interview): wire ambiguity milestones into prompt and MCP response (#363)

### PR DESCRIPTION
**Pull request by shaun0927**
## Summary

- Wire milestone context into `_build_ambiguity_snapshot_prompt()` so the LLM question generator adapts strategy per stage (broad questions at INITIAL → edge-case focus at REFINED)
- Add `"milestone"` field to all 5 MCP response `meta` blocks in `InterviewHandler` for downstream tooling and user-facing display
- Update `_format_question_with_ambiguity()` to show milestone inline: `(ambiguity: 0.44 [initial]) Question?`
- Update completion response `score_line` to include milestone: `(ambiguity: 0.18 [ready]) Ready for Seed generation.`

## Why this is needed

PR #379 defined the milestone data layer but didn't expose it anywhere. Without wiring, milestones are invisible to both users and agents.

This PR connects milestones to the two consumers that matter:

1. **LLM prompt** — The question generator now sees `"Milestone: **REFINED** — Success criteria partially defined. Edge cases and non-goals remain."` and can tailor its questioning strategy accordingly. Previously it only saw a raw number like `0.28`.

2. **MCP response meta** — Main sessions and downstream tools receive structured `"milestone": "refined"` in every response, enabling adaptive routing (e.g., auto-confirm decisions at INITIAL vs. precise edge-case questions at REFINED).

## Depends on

- #379 (`feat/ambiguity-milestones-data`) — milestone enum + helper functions

## Test plan

- [x] Updated 4 existing test assertions to match new display format
- [x] All 1078 unit tests pass (bigbang + mcp)
- [x] ruff check + format clean
- [x] New milestone prompt text verified in `test_interview.py`

### CODE REVIEW

File: src/ouroboros/bigbang/interview.py [hunk 1]
```diff
@@ -615,28 +615,42 @@ def _build_ambiguity_snapshot_prompt(self, state: InterviewState) -> str:
...
-            (
-                "- Closure mode active: yes"
-                if state.ambiguity_score <= SEED_CLOSER_ACTIVATION_THRESHOLD
-                else "- Closure mode active: no"
-            ),
-            (
-                "- Completion candidate streak: "
-                f"{state.completion_candidate_streak}/{AUTO_COMPLETE_STREAK_REQUIRED}"
-            ),
+            f"- Milestone: **{milestone.value.upper()}** — {milestone_desc}",
```
Comment (Q00) [inline src/ouroboros/bigbang/interview.py:628]: good

**Review by Q00 — APPROVED** [review]
